mlrun/projects/project.py

```diff
@@ -2292,7 +2293,7 @@ def create_remote(self, url, name="origin", branch=None):
         :param branch: Git branch to use as source
         """
         if not self.spec.repo:
-            raise ValueError("git repo is not set/defined")
+            self._ensure_git_repo()
```

Comment:
Can make it a bit cleaner if you would move the if above to inside the function.
So that inside the function you will have:
~~~
if self.spec.repo:
    return
~~~